sklearn/metrics/scorer.py

```diff
@@ -44,7 +45,54 @@
 from ..base import is_regressor
 
 
-class _BaseScorer(metaclass=ABCMeta):
+class _MultimetricScorer(dict):
+    """Callable dictionary for multimetric scoring."""
+
+    def __call__(self, estimator, *args, **kwargs):
+        """Evaluate predicted target values."""
+        scores = {}
+        cache = {} if self._use_cache() else None
+        method_cacher = partial(self._method_cacher, cache)
+
+        for name, scorer in self.items():
+            if isinstance(scorer, _BaseScorer):
+                score = scorer._score(estimator, *args, **kwargs,
+                                      method_cacher=method_cacher)
+            else:
+                score = scorer(estimator, *args, **kwargs)
+            scores[name] = score
+        return scores
+
+    def _use_cache(self):
+        """Return True if using a cache is desired."""
+        if len(self) == 1:
+            return False
+
+        counter = Counter([type(v) for v in self.values()])
+
+        for known_type in [_PredictScorer, _ProbaScorer, _ThresholdScorer]:
+            if counter[known_type] > 1:
+                return True
+
+        if counter[_ThresholdScorer] > 0 and (counter[_PredictScorer] or
+                                              counter[_ThresholdScorer]):
```

Comment:
This is equivalent to 

```
counter[_ThresholdScorer] and (counter[_PredictScorer]
```

(the `or` isn't useful)

Reply:
This should have been:

```
        if counter[_ThresholdScorer] and (counter[_PredictScorer] or
                                          counter[_ProbaScorer]):
```